src/server/mutate.rs

```diff
@@ -56,12 +102,30 @@ async fn mutate_internal(review: AdmissionReview) -> Result<impl warp::Reply, wa
         ));
     };
 
-    let patch = r#"[{
-        "op": "add",
-        "path": "/spec/runtimeClassName",
-        "value": "edera"
-    }]"#;
-    let patch = BASE64_STANDARD.encode(patch);
+    // Get name, falling back to generateName if name is missing.
+    let metadata = request.object.metadata;
+    let name = metadata.name.unwrap_or_else(|| {
+        metadata
+            .generate_name
+            .unwrap_or_else(|| "unknown".to_string())
+    });
+    let namespace = metadata.namespace;
+
+    // Determine patch path based on object kind. Default to pod.
+    let patch_path = if let Some(kind_info) = &request.kind {
+        match kind_info.kind.as_str() {
+            "Deployment" | "ReplicaSet" | "StatefulSet" => "/spec/template/spec/runtimeClassName",
```

Comment:
Should we add DaemonSet here too?

Reply:
Checked https://kubernetes.io/docs/concepts/workloads/controllers/daemonset/ and I'd agree!